lib/quicly.c

```diff
@@ -3129,12 +3147,31 @@ int quicly_send(quicly_conn_t *conn, quicly_datagram_t **packets, size_t *num_pa
         assert(conn->egress.last_retransmittable_sent_at == now);
     }
 
+#if 0
+    /* send probe if requested */
+    if (conn->super.peer.address_validation.send_probe && s.num_packets == 0) {
```

Comment:
you only want to do this if nothing has been sent in this round right?

Reply:
Yes and the condition is checked by `s.num_packets == 0`.

Note that the probe emission is now instructed by the last argument of `send_handshake_flow`.